src/js/http_outgoing.js

```diff
@@ -100,12 +100,12 @@ OutgoingMessage.prototype._send = function(chunk, encoding, callback) {
     callback = encoding;
   }
 
-  if (util.isBuffer(chunk)) {
-    chunk = chunk.toString();
-  }
-
   if (!this._sentHeader) {
-    chunk = this._header + '\r\n' + chunk;
+    if (util.isBuffer(chunk)) {
```

Comment:
Why do we need this distinction? Can't we just push the header to the `chunks` list if it was not sent out yet and keep the chunk as it is?